### PR TITLE
Fixed missing return in download_file, improve handling

### PR DESCRIPTION
**Pull request by medariox**
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

- [x] Fix torrent cache providers shuffle

### CODE REVIEW

File: medusa/helpers.py [hunk 1]
```diff
@@ -1256,20 +1263,23 @@ def download_file(url, filename, session=None, headers=None, **kwargs):
                             fp.flush()
 
                 chmodAsParent(filename)
-            except Exception:
-                logger.warning(u'Problem setting permissions or writing file to: {0}'.format(filename))
+            except Exception as e:
```
Comment (fernandog) [inline medusa/helpers.py]: @medariox what do you think about changing this to OSError?

File: medusa/providers/generic_provider.py [hunk 2]
```diff
@@ -119,15 +115,13 @@ def download_result(self, result):
...
             if download_file(url, filename, session=self.session, headers=self.headers,
                              hooks={'response': self.get_url_hook}, verify=verify):
+
                 if self._verify_download(filename):
-                    logger.log('Saved result to %s' % filename, logger.INFO)
+                    logger.log('Saved {result} to {location}'.format(result=result.name, location=filename))
                     return True
 
-                logger.log('Could not download %s' % url, logger.WARNING)
-                remove_file_failed(filename)
```
Comment (fernandog) [inline medusa/providers/generic_provider.py]: @medariox why are you removing this function?

Comment (medariox) [inline medusa/providers/generic_provider.py]: It's handled in _verify_download

File: medusa/helpers.py (hunk 1 above)
Comment (medariox) [inline medusa/helpers.py]: changed

**Comment by fernandog** [comment]
approved

**Pull request merged**